python/cugraph-dgl/examples/muti_trainer_MG_example/workflow.py

```diff
@@ -0,0 +1,233 @@
+    import cugraph_dgl
+
+    temp_dir = tempfile.TemporaryDirectory()
+    sampler = cugraph_dgl.dataloading.NeighborSampler([10, 20])
+    dataloader = cugraph_dgl.dataloading.DataLoader(
+        gs,
+        train_idx,
+        sampler,
+        sampling_output_dir=temp_dir.name,
+        batches_per_partition=10,
+        device=device,  # Put the sampled MFGs on CPU or GPU
+        use_ddp=True,  # Make it work with distributed data parallel
+        batch_size=1024,
+        shuffle=False,  # Whether to shuffle the nodes for every epoch
+        drop_last=False,
+        num_workers=0,
+    )
+    return dataloader
+
+
+def run_workflow(rank, devices, scheduler_address):
+    # Below sets gpu_num
+    dev_id = devices[rank]
+    initalize_pytorch_worker(dev_id)
+    device = torch.device(f"cuda:{dev_id}")
+    # cugraph dask client initialization
+    client = create_dask_client(scheduler_address)
+
+    # Pytorch training worker initialization
+    dist_init_method = "tcp://{master_ip}:{master_port}".format(
+        master_ip="127.0.0.1", master_port="12346"
+    )
+
+    torch.distributed.init_process_group(
+        backend="nccl",
+        init_method=dist_init_method,
+        world_size=len(devices),
+        rank=rank,
+    )
+
+    print(f"rank {rank}.", flush=True)
+    print("Initalized across GPUs.")
+
+    event = Dask_Event("cugraph_gs_creation_event")
+    if rank == 0:
+        (
+            gs,
+            train_idx,
+            valid_idx,
+            test_idx,
+        ) = create_cugraph_graphstore_from_dgl_dataset(
+            "ogbn-products", single_gpu=False
+        )
+        client.publish_dataset(cugraph_gs=gs)
+        client.publish_dataset(train_idx=train_idx)
+        client.publish_dataset(valid_idx=valid_idx)
+        client.publish_dataset(test_idx=test_idx)
+        event.set()
+    else:
+        if event.wait(timeout=1000):
+            gs = client.get_dataset("cugraph_gs")
+            train_idx = client.get_dataset("train_idx")
+            valid_idx = client.get_dataset("valid_idx")
+            test_idx = client.get_dataset("test_idx")
+        else:
+            raise RuntimeError(f"Fetch cugraph_gs to worker_id {rank} failed")
+
+    torch.distributed.barrier()
+    print(f"Loading cugraph_store to worker {rank} is complete", flush=True)
+    dataloader = create_dataloader(gs, train_idx, device)
+    print("Data Loading Complete", flush=True)
+    del gs  # Clean up gs reference
+    # Comment below
+    st = time.time()
+    for step, (input_nodes, seeds, blocks) in enumerate(dataloader):
+        pass
+    et = time.time()
+    print(f"Warmup loading took = {et-st} s on worker = {rank}")
+    torch.distributed.barrier()
+
+    n_epochs = 30
+    total_st = time.time()
+    for i in range(0, n_epochs):
+        st = time.time()
+        for step, (input_nodes, seeds, blocks) in enumerate(dataloader):
+            pass
+            # print(len(input_nodes))
```

Comment:
I notice you're not actually doing model training here, that's intentional, right?

Reply:
ok